pilot/pkg/controllers/ipallocate/ipallocate.go

```diff
@@ -410,6 +415,11 @@ func (c *IPAllocator) statusPatchForAddresses(se *networkingv1.ServiceEntry, for
 			Path:      "/status/addresses",
 			Value:     assignedAddresses,
 		},
+		{
+			Operation: "add",
+			Path:      "/status/observedGeneration",
```

Comment:
This field is not meant to be used this way. if you read the documentation, it specifically says "This indicates the generation the `Reconciled` condition refers to". 

It is a bad API as well, which it is why it was deprecated and moved to tying the observedGeneration to the individual condition instead of at the top level -- as there can be multiple writers of status.